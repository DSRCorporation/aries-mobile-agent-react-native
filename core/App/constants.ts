import { PredicateType } from '@aries-framework/core'

import { ProofRequestTemplate, ProofRequestType } from './types/proof-reqeust-template'
import { PINValidationRules } from './types/security'

export const defaultLanguage = 'en'
export const dateIntFormat = 'YYYYMMDD'

const lengthOfhiddenAttributes = 10
const unicodeForBulletCharacter = '\u2022'
export const hiddenFieldValue = Array(lengthOfhiddenAttributes).fill(unicodeForBulletCharacter).join('')
// Used to property prefix TestIDs so they can be looked up
// by on-device automated testing systems like SauceLabs.
export const testIdPrefix = 'com.ariesbifold:id/'

export enum LocalStorageKeys {
  Onboarding = 'OnboardingState',
  LoginAttempts = 'LoginAttempts',
  // FIXME: Once hooks are updated this should no longer be necessary
  RevokedCredentials = 'RevokedCredentials',
  RevokedCredentialsMessageDismissed = 'RevokedCredentialsMessageDismissed',
  Preferences = 'PreferencesState',
}

export enum KeychainServices {
  Salt = 'secret.wallet.salt',
  Key = 'secret.wallet.key',
}

export enum EventTypes {
  ERROR_ADDED = 'ErrorAdded',
  ERROR_REMOVED = 'ErrorRemoved',
  BIOMETRY_UPDATE = 'BiometryUpdate',
  BIOMETRY_ERROR = 'BiometryError',
}

export const second = 1000
export const minute = 60000
export const hour = 3600000

// wallet timeout of 5 minutes, 0 means the wallet never locks due to inactivity
export const walletTimeout = minute * 5

/* lockout attempt rules: The base rules apply the lockout at a specified number of incorrect attempts,
 and the threshold rules apply the lockout penalty to each attempt after the threshold that falls on the attemptIncrement.
 (In this case the threshold rule applies to every 5th incorrect login after 20)
5 incorrect => 1 minute lockout
10 incorrect => 10 minute lockout
15 incorrect => 1 hour lockout
20, 25, 30, etc incorrect => 1 day lockout
*/
export const attemptLockoutBaseRules: Record<number, number | undefined> = {
  5: minute,
  10: 10 * minute,
  15: hour,
}
export const attemptLockoutThresholdRules = {
  attemptThreshold: 20,
  attemptIncrement: 5,
  attemptPenalty: 24 * hour,
}

export const walletId = 'walletId'

export const minPINLength = 6

export const PINRules: PINValidationRules = {
  only_numbers: true,
  min_length: 6,
  max_length: 6,
  no_repeated_numbers: false,
  no_repetition_of_the_two_same_numbers: false,
  no_series_of_numbers: false,
  no_even_or_odd_series_of_numbers: false,
  no_cross_pattern: false,
}

export const defaultProofRequestTemplates: Array<ProofRequestTemplate> = [
  {
    id: '8a83675e-f864-4e5a-9c4d-9787f1034c04',
    title: 'Full name',
    details: 'Verify the full name of a person',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedAttributes: [
            {
              name: 'given_names',
            },
            {
              name: 'family_name',
            },
          ],
        },
      ],
    },
  },
  {
    id: '3339c1e1-681f-433c-9f2f-8d0d1af0b3d2',
    title: '19+ and Full name',
    details: 'Verify if a person is 19 years end up and full name.',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedAttributes: [
            {
              name: 'given_names',
            },
            {
              name: 'family_name',
            },
          ],
          requestedPredicates: [
            {
              name: 'age',
              predicateType: PredicateType.GreaterThanOrEqualTo,
              predicateValue: 19,
            },
          ],
        },
      ],
    },
  },
  {
    id: '18183e86-4528-4d10-80f8-aa5c7cbbbfe7',
    title: 'Over 19 years of age',
    details: 'Verify if a person is 19 years end up.',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedPredicates: [
            {
              name: 'age',
              predicateType: PredicateType.GreaterThanOrEqualTo,
              predicateValue: 19,
            },
          ],
        },
      ],
    },
  },
  {
    id: 'd91c7aab-af1a-42b6-8f1f-2702ed6e1f98',
    title: 'Practising lawyer',
    details: 'Verify if a person`is a practicing lawyer.',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Practising Lawyer',
          requestedAttributes: [
            {
              names: ['given_names', 'family_name', 'ppid', 'practicing_status'],
            },
          ],
        },
      ],
    },
  },
  {
    id: 'e4cf6f7c-5abe-4e0d-b299-f944fe5ef8b2',
    title: 'Practising lawyer and full name',
    details: 'Verify if a person`is a practicing lawyer using two different credentials for extra assuarnce',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedAttributes: [
            {
              name: 'given_names',
            },
            {
              name: 'family_name',
            },
          ],
        },
        {
          schema: 'Practising Lawyer',
          requestedAttributes: [
            {
              names: ['given_names', 'family_name', 'ppid', 'practicing_status'],
            },
          ],
        },
      ],
    },
  },
  {
    id: '18183e86-4528-4d10-80f8-aa5c7cbbbfe7',
    title: 'Over some years of age',
    details: 'Verify if a person is over some years ends up.',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedPredicates: [
            {
              name: 'age',
              predicateType: PredicateType.GreaterThanOrEqualTo,
              predicateValue: 19,
              parameterizable: true,
            },
          ],
        },
      ],
    },
  },
]
