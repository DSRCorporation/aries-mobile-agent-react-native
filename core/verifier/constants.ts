import { PredicateType } from '@aries-framework/core'

import { ProofRequestTemplate, ProofRequestType } from './types/proof-reqeust-template'

export const defaultProofRequestTemplates: Array<ProofRequestTemplate> = [
  {
    id: '8a83675e-f864-4e5a-9c4d-9787f1034c04',
    title: 'Full name',
    details: 'Verify the full name of a person',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedAttributes: [
            {
              name: 'given_names',
            },
            {
              name: 'family_name',
            },
            {
              names: ['issued', 'country'],
            },
          ],
          requestedPredicates: [
            {
              name: 'age',
              predicateType: PredicateType.GreaterThanOrEqualTo,
              predicateValue: 17,
            },
          ],
        },
      ],
    },
  },
  {
    id: '3339c1e1-681f-433c-9f2f-8d0d1af0b3d2',
    title: '19+ and Full name',
    details: 'Verify if a person is 19 years end up and full name.',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedAttributes: [
            {
              name: 'given_names',
            },
            {
              name: 'family_name',
            },
          ],
          requestedPredicates: [
            {
              name: 'age',
              predicateType: PredicateType.GreaterThanOrEqualTo,
              predicateValue: 19,
            },
          ],
        },
      ],
    },
  },
  {
    id: '18183e86-4528-4d10-80f8-aa5c7cbbbfe7',
    title: 'Over 19 years of age',
    details: 'Verify if a person is 19 years end up.',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedPredicates: [
            {
              name: 'age',
              predicateType: PredicateType.GreaterThanOrEqualTo,
              predicateValue: 19,
            },
          ],
        },
      ],
    },
  },
  {
    id: 'd91c7aab-af1a-42b6-8f1f-2702ed6e1f98',
    title: 'Practising lawyer',
    details: 'Verify if a person`is a practicing lawyer.',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Practising Lawyer',
          requestedAttributes: [
            {
              names: ['given_names', 'family_name', 'ppid', 'practicing_status'],
            },
          ],
        },
      ],
    },
  },
  {
    id: 'e4cf6f7c-5abe-4e0d-b299-f944fe5ef8b2',
    title: 'Practising lawyer and full name',
    details: 'Verify if a person`is a practicing lawyer using two different credentials for extra assuarnce',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedAttributes: [
            {
              name: 'given_names',
            },
            {
              name: 'family_name',
            },
          ],
        },
        {
          schema: 'Practising Lawyer',
          requestedAttributes: [
            {
              names: ['given_names', 'family_name', 'ppid', 'practicing_status'],
            },
          ],
        },
      ],
    },
  },
  {
    id: '18183e86-4528-4d10-80f8-aa5c7cbbbfe7',
    title: 'Over some years of age',
    details: 'Verify if a person is over some years ends up.',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedPredicates: [
            {
              name: 'age',
              predicateType: PredicateType.GreaterThanOrEqualTo,
              predicateValue: 19,
              parameterizable: true,
            },
          ],
        },
      ],
    },
  },
]
