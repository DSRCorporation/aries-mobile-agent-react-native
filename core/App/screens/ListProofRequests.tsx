import { StackNavigationProp } from '@react-navigation/stack'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'

import { defaultProofRequestTemplates } from '../constants'
import { ColorPallet } from '../theme'
import { ProofRequestsStackParams, Screens } from '../types/navigators'
import { ProofRequestTemplate, ProofRequestType } from '../types/proof-reqeust-template'

interface ListProofRequestsProps {
  navigation: StackNavigationProp<ProofRequestsStackParams>
}

const style = StyleSheet.create({
  container: {
    margin: 24,
  },
  proofButton: {
    flexDirection: 'row',
    borderRadius: 8,
    padding: 12,
    backgroundColor: '#fff',
    marginBottom: 10,
  },
  textContainer: {
    flex: 1,
  },
  proofTitle: {
    fontWeight: 'bold',
    marginBottom: 6,
  },
  proofCaption: {
    color: ColorPallet.grayscale.mediumGrey,
    fontSize: 11,
  },
  proofText: {
    color: ColorPallet.grayscale.black,
  },
  proofDetails: {
    marginBottom: 4,
  },
  iconContainer: {
    alignSelf: 'center',
  },
  icon: {
    color: ColorPallet.grayscale.black,
  },
})

const iconSize = 35
const iconName = 'chevron-right'

const ListProofRequests: React.FC<ListProofRequestsProps> = ({ navigation }) => {
  const { t } = useTranslation()
  const records = defaultProofRequestTemplates
  const hasPredicates = (record: ProofRequestTemplate): boolean => {
    const indyPayload = record.payload.type === ProofRequestType.Indy ? record.payload : null
    if (!indyPayload) return false
    return indyPayload.data.some((d) => d.requestedPredicates && d.requestedPredicates?.length > 0)
  }
  return (
    <View style={style.container}>
      {records.map((record) => (
        <TouchableOpacity style={style.proofButton} onPress={() => navigation.navigate(Screens.ProofRequestFullName)}>
          <View style={style.textContainer}>
            <Text style={[style.proofText, style.proofTitle]} numberOfLines={1}>
              {record.title}
            </Text>
            <Text style={[style.proofText, style.proofDetails]} numberOfLines={2}>
              {record.details}
            </Text>
            {hasPredicates(record) && (
              <Text style={[style.proofText, style.proofCaption]}>{t('ProofRequest.ZeroKnowledgeProof')}</Text>
            )}
          </View>
          <View style={style.iconContainer}>
            <Icon style={style.icon} size={iconSize} name={iconName} />
          </View>
        </TouchableOpacity>
      ))}
    </View>
  )
}

export default ListProofRequests
