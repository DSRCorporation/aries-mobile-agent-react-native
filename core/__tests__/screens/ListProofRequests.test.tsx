import mockRNCNetInfo from '@react-native-community/netinfo/jest/netinfo-mock'
import { useNavigation } from '@react-navigation/core'
import { act, fireEvent, render } from '@testing-library/react-native'
import React from 'react'

import { ConfigurationContext } from '../../App/contexts/configuration'
import { NetworkProvider } from '../../App/contexts/network'
import configurationContext from '../contexts/configuration'
import ListProofRequests from '../../App/screens/ListProofRequests'
import { defaultProofRequestTemplates } from '../../verifier/constants'
import ProofRequestDetails from '../../App/screens/ProofRequestDetails'

jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter')
jest.mock('@react-native-community/netinfo', () => mockRNCNetInfo)
jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})
// eslint-disable-next-line @typescript-eslint/no-empty-function
jest.mock('react-native-localize', () => {})
jest.useFakeTimers('legacy')
jest.spyOn(global, 'setTimeout')

const navigation = useNavigation()

describe('ListProofRequests Component', () => {
  const renderView = (params?: {}) => {
    return render(
      <ConfigurationContext.Provider value={configurationContext}>
        <NetworkProvider>
          <ListProofRequests navigation={navigation as any} route={{ params: params || {} } as any} />
        </NetworkProvider>
      </ConfigurationContext.Provider>
    )
  }

  test('Renders correctly', async () => {
    const tree = renderView()
    await act(async () => null)
    expect(tree).toMatchSnapshot()
  })

  test('Schema names are human readable', async () => {
    const tree = renderView()

    const fullName = await tree.findByText('19+ and Full name', { exact: false })
    const overYearsOfAge = await tree.findByText(' 19 years of age', { exact: false })

    expect(fullName).not.toBe(null)
    expect(overYearsOfAge).not.toBe(null)
  })

  test('Pressing on a request template takes the user to a proof request template detail screen', async () => {
    const tree = renderView()

    await act(async () => {
      const templateItemInstances = await tree.findAllByText('19+ and Full name', { exact: false })

      expect(templateItemInstances.length).toBe(1)

      const templateItemInstance = templateItemInstances[0]

      fireEvent(templateItemInstance, 'press')

      expect(navigation.navigate).toBeCalledWith('Proof Request Details', {
        templateId: defaultProofRequestTemplates[1].id,
      })
    })
  })
})
