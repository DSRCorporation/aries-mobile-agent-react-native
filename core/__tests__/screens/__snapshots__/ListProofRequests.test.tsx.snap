// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListProofRequests Component Renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Object {
      "elevation": 5,
      "flexGrow": 1,
      "margin": 24,
    }
  }
>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    data={
      Array [
        Object {
          "description": "Verify the full name of a person",
          "id": "BC:5:FullName:0.0.1:indy",
          "name": "Full name",
          "payload": Object {
            "data": Array [
              Object {
                "requestedAttributes": Array [
                  Object {
                    "name": "given_names",
                    "restrictions": Array [
                      Object {
                        "schema_id": "XUxBrVSALWHLeycAUhrNr9:2:Person:1.0",
                      },
                    ],
                  },
                  Object {
                    "name": "family_name",
                    "restrictions": Array [
                      Object {
                        "schema_id": "XUxBrVSALWHLeycAUhrNr9:2:Person:1.0",
                      },
                    ],
                  },
                ],
                "schema": "XUxBrVSALWHLeycAUhrNr9:2:Person:1.0",
              },
            ],
            "type": "indy",
          },
          "version": "0.0.1",
        },
        Object {
          "description": "Verify if a person is 19 years end up and full name.",
          "id": "BC:5:19+AndFullName:0.0.1:indy",
          "name": "19+ and Full name",
          "payload": Object {
            "data": Array [
              Object {
                "requestedAttributes": Array [
                  Object {
                    "names": Array [
                      "given_names",
                      "family_name",
                    ],
                    "restrictions": Array [
                      Object {
                        "schema_id": "XUxBrVSALWHLeycAUhrNr9:2:Person:1.0",
                      },
                    ],
                  },
                ],
                "requestedPredicates": Array [
                  Object {
                    "name": "birthdate_dateint",
                    "predicateType": ">=",
                    "predicateValue": 18,
                    "restrictions": Array [
                      Object {
                        "schema_id": "XUxBrVSALWHLeycAUhrNr9:2:Person:1.0",
                      },
                    ],
                  },
                ],
                "schema": "XUxBrVSALWHLeycAUhrNr9:2:Person:1.0",
              },
            ],
            "type": "indy",
          },
          "version": "0.0.1",
        },
        Object {
          "description": "Verify if a person is 19 years end up.",
          "id": "BC:5:Over19YearsOfAge:0.0.1:indy",
          "name": "Over 19 years of age",
          "payload": Object {
            "data": Array [
              Object {
                "requestedPredicates": Array [
                  Object {
                    "name": "birthdate_dateint",
                    "predicateType": ">=",
                    "predicateValue": 18,
                    "restrictions": Array [
                      Object {
                        "schema_id": "XUxBrVSALWHLeycAUhrNr9:2:Person:1.0",
                      },
                    ],
                  },
                ],
                "schema": "XUxBrVSALWHLeycAUhrNr9:2:Person:1.0",
              },
            ],
            "type": "indy",
          },
          "version": "0.0.1",
        },
        Object {
          "description": "Verify if a person\`is a practicing lawyer.",
          "id": "BC:5:PractisingLawyer:0.0.1:indy",
          "name": "Practising lawyer",
          "payload": Object {
            "data": Array [
              Object {
                "requestedAttributes": Array [
                  Object {
                    "names": Array [
                      "Given Name",
                      "Surname",
                      "PPID",
                      "Member Status",
                    ],
                    "restrictions": Array [
                      Object {
                        "schema_id": "XUxBrVSALWHLeycAUhrNr9:2:Member Card:1.5.1",
                      },
                    ],
                  },
                ],
                "schema": "XUxBrVSALWHLeycAUhrNr9:2:Member Card:1.5.1",
              },
            ],
            "type": "indy",
          },
          "version": "0.0.1",
        },
        Object {
          "description": "Verify if a person\`is a practicing lawyer using two different credentials for extra assurance",
          "id": "BC:5:PractisingLawyerAndFullName:0.0.1:indy",
          "name": "Practising lawyer and full name",
          "payload": Object {
            "data": Array [
              Object {
                "requestedAttributes": Array [
                  Object {
                    "names": Array [
                      "given_names",
                      "family_name",
                    ],
                    "restrictions": Array [
                      Object {
                        "schema_id": "XUxBrVSALWHLeycAUhrNr9:2:Person:1.0",
                      },
                    ],
                  },
                ],
                "schema": "XUxBrVSALWHLeycAUhrNr9:2:Person:1.0",
              },
              Object {
                "requestedAttributes": Array [
                  Object {
                    "names": Array [
                      "Given Name",
                      "Surname",
                      "PPID",
                      "Member Status",
                    ],
                    "restrictions": Array [
                      Object {
                        "schema_id": "XUxBrVSALWHLeycAUhrNr9:2:Member Card:1.5.1",
                      },
                    ],
                  },
                ],
                "schema": "XUxBrVSALWHLeycAUhrNr9:2:Member Card:1.5.1",
              },
            ],
            "type": "indy",
          },
          "version": "0.0.1",
        },
        Object {
          "description": "Verify if a person is over some years ends up.",
          "id": "BC:5:OverSomeYearsOfAge:0.0.1:indy",
          "name": "Over some years of age",
          "payload": Object {
            "data": Array [
              Object {
                "requestedPredicates": Array [
                  Object {
                    "name": "birthdate_dateint",
                    "parameterizable": true,
                    "predicateType": ">=",
                    "predicateValue": 18,
                    "restrictions": Array [
                      Object {
                        "schema_id": "XUxBrVSALWHLeycAUhrNr9:2:Person:1.0",
                      },
                    ],
                  },
                ],
                "schema": "XUxBrVSALWHLeycAUhrNr9:2:Person:1.0",
              },
            ],
            "type": "indy",
          },
          "version": "0.0.1",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 8,
              "flexDirection": "row",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 12,
            }
          }
          testID="com.ariesbifold:id/ProofRequestsCard"
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 4,
                }
              }
            >
              Full name
            </Text>
            <Text
              numberOfLines={2}
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 16,
                  "fontWeight": "normal",
                  "marginBottom": 4,
                }
              }
            >
              Verify the full name of a person
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 36,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 8,
              "flexDirection": "row",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 12,
            }
          }
          testID="com.ariesbifold:id/ProofRequestsCard"
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 4,
                }
              }
            >
              19+ and Full name
            </Text>
            <Text
              numberOfLines={2}
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 16,
                  "fontWeight": "normal",
                  "marginBottom": 4,
                }
              }
            >
              Verify if a person is 19 years end up and full name.
            </Text>
            <Text
              style={
                Object {
                  "color": "#606060",
                  "fontSize": 12,
                }
              }
            >
              Verifier.ZeroKnowledgeProof
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 36,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 8,
              "flexDirection": "row",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 12,
            }
          }
          testID="com.ariesbifold:id/ProofRequestsCard"
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 4,
                }
              }
            >
              Over 19 years of age
            </Text>
            <Text
              numberOfLines={2}
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 16,
                  "fontWeight": "normal",
                  "marginBottom": 4,
                }
              }
            >
              Verify if a person is 19 years end up.
            </Text>
            <Text
              style={
                Object {
                  "color": "#606060",
                  "fontSize": 12,
                }
              }
            >
              Verifier.ZeroKnowledgeProof
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 36,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 8,
              "flexDirection": "row",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 12,
            }
          }
          testID="com.ariesbifold:id/ProofRequestsCard"
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 4,
                }
              }
            >
              Practising lawyer
            </Text>
            <Text
              numberOfLines={2}
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 16,
                  "fontWeight": "normal",
                  "marginBottom": 4,
                }
              }
            >
              Verify if a person\`is a practicing lawyer.
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 36,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 8,
              "flexDirection": "row",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 12,
            }
          }
          testID="com.ariesbifold:id/ProofRequestsCard"
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 4,
                }
              }
            >
              Practising lawyer and full name
            </Text>
            <Text
              numberOfLines={2}
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 16,
                  "fontWeight": "normal",
                  "marginBottom": 4,
                }
              }
            >
              Verify if a person\`is a practicing lawyer using two different credentials for extra assurance
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 36,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 8,
              "flexDirection": "row",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 12,
            }
          }
          testID="com.ariesbifold:id/ProofRequestsCard"
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 4,
                }
              }
            >
              Over some years of age
            </Text>
            <Text
              numberOfLines={2}
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 16,
                  "fontWeight": "normal",
                  "marginBottom": 4,
                }
              }
            >
              Verify if a person is over some years ends up.
            </Text>
            <Text
              style={
                Object {
                  "color": "#606060",
                  "fontSize": 12,
                }
              }
            >
              Verifier.ZeroKnowledgeProof
            </Text>
            <Text
              style={
                Object {
                  "color": "#606060",
                  "fontSize": 12,
                }
              }
            >
              Verifier.Parameterizable
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 36,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
